### PR TITLE
packets2: Add register and subscribe-related packets

### PR DESCRIPTION
**Pull request by mprymek**
PR adds register and subscribe-related packets: `Register`, `Regack`, `Subscribe`, `Suback`, `Unsubscribe`, `Unsuback`.

Failed tests are OK. They will pass as long as all packets are merged in.

### CODE REVIEW

**Review by dmajda — CHANGES_REQUESTED** [review]
Looks good overall, I have just two minor comments.

I have a more general question though: Handling of Topic Alias Type, Topic Alias, Topic Name and Topic Filter is done in multiple packets. The code dealing with packing, unpacking, computing packet length, and stringification is highly repetitive. Have you considered abstracting it a bit?

File: packets2/regack.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		p.TopicAliasType, p.TopicAlias, p.ReasonCode, p.packetID)
+}
+
+func (p *Regack) encodeFlags() byte {
+	return byte(p.TopicAliasType) & flagsTopicAliasTypeBits
+}
+
+func (p *Regack) decodeFlags(b byte) {
+	p.TopicAliasType = TopicAliasType(b & flagsTopicAliasTypeBits)
+}
```
[the remark this answers is not in the document]
Comment (mprymek) [reply packets2/regack.go:70]: ok

**Review by dmajda — COMMENTED** [review]
Changes look good. I’m still curious about the answer to my question [above](https://github.com/energomonitor/bisquitt/pull/58#pullrequestreview-1173557558):

> I have a more general question though: Handling of Topic Alias Type, Topic Alias, Topic Name and Topic Filter is done in multiple packets. The code dealing with packing, unpacking, computing packet length, and stringification is highly repetitive. Have you considered abstracting it a bit?

**Comment by mprymek** [comment]
> Changes look good. I’m still curious about the answer to my question [above](https://github.com/energomonitor/bisquitt/pull/58#pullrequestreview-1173557558):
> 
> > I have a more general question though: Handling of Topic Alias Type, Topic Alias, Topic Name and Topic Filter is done in multiple packets. The code dealing with packing, unpacking, computing packet length, and stringification is highly repetitive. Have you considered abstracting it a bit?

You are right that handling of these properties is much more unified and sane in MQTT-SN v.2 than in v.1 (e.g. the "long" topic ID type has the same flag value as the registered topic ID type in v.1 [1]) and there's probably room for some abstraction. Nevertheless, there are some minor deviations in particular packets and I must carefully investigate if it's  worth it and the abstraction will not just complicate the code without much benefit.

I think it's out of scope of this PR. I made an issue [2] and will take a look later.

[1] https://github.com/energomonitor/bisquitt/blob/main/packets1/packets1.go#L44
[2] https://github.com/energomonitor/bisquitt/issues/64

**Review by dmajda — APPROVED** [review]

**Pull request merged**